app/lib/package/backend.dart

```diff
@@ -925,6 +925,10 @@ Future<_ValidatedUpload> _parseAndValidateUpload(
     throw GenericProcessingException(
         'Unable to canonicalize the version: ${pubspec.version}');
   }
+  if (versionString.endsWith('tar.gz')) {
```

Comment:
I'm not convinced this is a good idea.. Also you can settle with `.tar.gz`

Reply:
I guess there is little harm in this. but it just seems like an arbitrary restriction...

Reply:
If we want to go down this route a means of hardening against various hacks, I suggest we forbid all common file extensions, but really don't we have to trust the users of versions numbers read the spec for the format that they can have.

Reply:
I came up with this while trying to fix the version regexp. It turned out to be simpler than I've originally thought, so this may not be needed at all.